src/App.tsx

```diff
@@ -1,56 +1,81 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
+import cn from 'classnames';
 import { peopleFromServer } from './data/people';
+import { DropDownMenu } from './components/DropDownMenu/DropDownMenu';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedPersonSlug, setSelectedPersonSlug] = useState('');
+  const [isSugActive, setIsSugActive] = useState(false);
+
+  const visiblePeople = useMemo(() => {
+    const normalizedQuery = appliedQuery.toLowerCase().trim();
+
+    if (appliedQuery) {
+      setIsSugActive(true);
+    }
+
+    if (!appliedQuery) {
+      setSelectedPersonSlug('');
+      setIsSugActive(false);
+    }
+
+    return peopleFromServer.filter(person => (
```

Comment:
you can just check if you have appliedQuery then filter. In opposite way just return an empty array